enumerable/concat.go

```diff
@@ -1,5 +1,15 @@
 package enumerable
 
+// Concatenation is an extention of the enumerable interface allowing new sources
+// to be added after initial construction.
+type Concatenation[T any] interface {
```

Comment:
`expanded` or `Dynamic` is way too vague for my liking as it can mean a whole bunch of stuff.  And concat/concatenation is a fairly common term for this behaviour.